include/networkit/distance/STSP.hpp

```diff
@@ -0,0 +1,111 @@
+/*
+ * STSP.hpp
+ *
+ *  Created on: 14.06.2019
+ *      Author: Eugenio Angriman <angrimae@hu-berlin.de>
+ */
+
+#ifndef STSP_H
+#define STSP_H
+
+#include <networkit/Globals.hpp>
+#include <networkit/base/Algorithm.hpp>
+#include <networkit/graph/Graph.hpp>
+
+namespace NetworKit {
+
+/**
+ * @ingroup distance
+ * Abstract base class for source-target shortest path algorithms.
+ */
+class STSP : public Algorithm {
+
+public:
+    /**
+     * Creates the STSP class for a graph @a G, source node @a source, and
+     * target node @a target.
+     *
+     * @param G The graph.
+     * @param source The source node.
+     * @param target The target node.
+     * @param storePred If true, the algorithm will also store the predecessors
+     * and reconstruct a shortest path from @a source and @a target.
+     */
+    STSP(const Graph &G, node source, node target, bool storePred = true)
+        : G(G), source(source), target(target), storePred(storePred) {
+        if (!G.hasNode(source))
+            throw std::runtime_error("Error: source node not in the graph!");
+        if (!G.hasNode(target))
+            throw std::runtime_error("Error: target node not in the graph!");
+        if (source == target)
+            INFO("Source and target nodes are equal!");
+    }
+
+    /**
+     * Returns a shortest path from the source node to the target node (without
+     * including them). Note: the shortest path can be constructed only if the
+     * algorithm is executed with @a storePred set to true.
+     *
+     * @return A shortest path from the @a source to @a target.
+     */
+    std::vector<node> getPath() const {
+        checkStoreDistances();
+        assureFinished();
+        return path;
+    }
+
+    /**
+     * Returns the predecessor nodes from the @a target to the source. Note:
+     * predecessors are stored only if the algorithm is executed with @a
+     * storePred set to true.
+     *
+     * @return The list of predecessors from @a target to @a source.
+     */
+    std::vector<node> getPredecessors() const {
+        checkStoreDistances();
+        assureFinished();
+        return pred;
+    }
+
+protected:
+    const Graph &G;
+    node source, target;
+    const bool storePred;
+    std::vector<uint8_t> visited;
+    uint8_t ts = 0;
+    static constexpr uint8_t ballMask = uint8_t(1) << 7;
+    std::vector<node> pred;
+    std::vector<node> path;
+
+    void buildPath();
```

Comment:
What is the intended use case for that function? Derived classes call it to reconstruct a path? Add a comment to make that clear.